src/geo/leaflet/leaflet-layer-view-factory.js

```diff
@@ -1,24 +1,54 @@
+var _ = require('underscore');
 var log = require('cdb.log');
 var LeafletTiledLayerView = require('./leaflet-tiled-layer-view');
 var LeafletWMSLayerView = require('./leaflet-wms-layer-view');
 var LeafletPlainLayerView = require('./leaflet-plain-layer-view');
 var LeafletCartoDBLayerGroupView = require('./leaflet-cartodb-layer-group-view');
 var LeafletTorqueLayerView = require('./leaflet-torque-layer-view');
 var LeafletCartoDBWebglLayerGroupView = require('./leaflet-cartodb-webgl-layer-group-view');
+var TangramCartoCSS = require('tangram-cartocss');
+var RenderModes = require('../../vis/render-modes');
 
-var LayerGroupViewConstructor = function (layerGroupModel, mapModel, options) {
-  if (options.vector) {
-    return new LeafletCartoDBWebglLayerGroupView(layerGroupModel, mapModel);
+var MAX_NUMBER_OF_FEATURES_FOR_WEBGL = 1000;
+
+var LayerGroupViewConstructor = function (layerGroupModel, nativeMap, mapModel, settingsModel) {
+  if (canMapBeRenderedClientSide(mapModel, settingsModel)) {
+    return new LeafletCartoDBWebglLayerGroupView(layerGroupModel, nativeMap);
+  }
+
+  return new LeafletCartoDBLayerGroupView(layerGroupModel, nativeMap);
+};
+
+var canMapBeRenderedClientSide = function (mapModel, settingsModel) {
+  var mapRenderMode = settingsModel.get('renderMode');
+
+  if (mapRenderMode === RenderModes.VECTOR) {
+    return true;
   }
-  return new LeafletCartoDBLayerGroupView(layerGroupModel, mapModel);
+
+  if (mapRenderMode === RenderModes.RASTER) {
+    return false;
+  }
+
+  // mapRenderMode === RenderModes.AUTO
+  return mapModel.getNumberOfFeatures() < MAX_NUMBER_OF_FEATURES_FOR_WEBGL &&
```

Comment:
:point_up:  @donflopez @javisantana here's the code that decides whether to use vector or raster when renderMode is auto.